tracer/test/Datadog.Trace.Tests/Util/Delegates/FuncInstrumentationTests.cs

```diff
@@ -479,6 +479,126 @@ public async Task Async1Test()
         value.Should().Be(4);
     }
 
+    [Fact]
+    public async Task Async1WithAsyncExceptionTest()
+    {
+        var result = new StrongBox<int>(0);
+        var callbacks = new Async1Callbacks();
+        Func<string, Task<int>> func = async (arg1) =>
+        {
+            callbacks.Count.Value++;
+            // force an exception
+            int x = 0, y = 0, z = 0;
+            z = x / y;
+            await Task.Yield();
+            return 42;
+        };
+        func = func.Instrument(callbacks);
+        await Assert.ThrowsAsync<DivideByZeroException>(
+            async () =>
+            {
+                result = new StrongBox<int>(await func("Arg01").ConfigureAwait(false));
+            }).ConfigureAwait(false);
+
+        callbacks.Count.Value.Should().Be(4);
```

Comment:
So just to make sure I'm understanding this part of the unit test, the result is 4 because we're calling Count.Value++ in the following sequence right?
- OnDelegateBegin
- original `func` delegate
- ⚠️ Exception happens ⚠️ 
- OnDelegateEnd
- OnDelegateEndAsync

Reply:
Yes the execution order will depend id the target delegate has finished or not. If when we are setting the continuation the task has been completed we call the continuation directly. If not we set the continuation so the execution can occur after the OnDelegateEnd. In the case of an exception because the initial Task is completed (faulted) we run the EndAsync before End.